tmk/cpp/bin/tmk-query-with-faiss.cpp

```diff
@@ -168,16 +166,29 @@ int main(int argc, char** argv) {
   if (num_centroids > num_database_vectors) {
     num_centroids = num_database_vectors;
   }
-  printf("VECTOR_DIMENSION     %d\n", (int)vector_dimension);
-  printf("NUM_DATABASE_VECTORS %d\n", (int)num_database_vectors);
-  printf("NUM_CENTROIDS        %d\n", (int)num_centroids);
+  if (verbose) {
+    printf("VECTOR_DIMENSION     %d\n", (int)vector_dimension);
+    printf("NUM_DATABASE_VECTORS %d\n", (int)num_database_vectors);
+    printf("NUM_CENTROIDS        %d\n", (int)num_centroids);
+  }
 
   // The coarse quantizer should not be deallocated before the index.
   // 4 = number of bytes per code (vector_dimension must be a multiple of this)
   // 8 = number of bits per sub-code (almost always 8)
   faiss::IndexIVFPQ faiss_index(&coarse_quantizer, vector_dimension, num_centroids, 4, 8);
```

Comment:
I'm confused about why this isn't inner product, since that's what is needed to query cosin distance. Do you know?

Ref: https://github.com/facebookresearch/faiss/wiki/Faiss-indexes

Reply:
@Dcallies I found the same references as you, and had the exact same set of thoughts. To query cosine similarity, it seems like we're supposed to use IP and L2-normalized vectors. I just pushed the changes I mentioned at the end of https://github.com/facebook/ThreatExchange/pull/959#issuecomment-1079741989: [a5863bd](https://github.com/facebook/ThreatExchange/pull/959/commits/a5863bddadd91b5c1308ed0e02a482cadd6517e7). Hopefully, it's more clear.

I do see an improved result set, but I have no idea if I've called `fvec_renorm_L2` correctly. I don't see any errors though.